app/src/components/assets/Asset.tsx

```diff
@@ -0,0 +1,165 @@
+import { faBan, faFlag } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IonCard, IonCardContent, IonCardHeader, IonCardSubtitle, IonCardTitle, IonCol, IonContent, IonItem, IonLabel, IonList, IonRow, useIonViewWillLeave } from "@ionic/react";
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import Api from "../../controllers/Api";
+import { s3url, fileExtensionToIcon, formatSize } from "../../globals/functions";
+import { AssetData, AssetTypeData } from "../../globals/interfaces";
+import Page from "../../pages/Page";
+
+const Asset = () => {
+    let { type, asset } = useParams<{type: string, asset: string}>();
+    const [assetType, setAssetType] = useState<AssetTypeData>({tags: [], thumnails: [], files: [], fields: []});
+    const [thisAsset, setThisAsset] = useState<AssetData>({flagsblocks: {BLOCK:[], FLAG:[], COUNT:{BLOCK: 0, FLAG: 0}}});
+    let cancelToken = axios.CancelToken.source();
+
+    //get data
+    useEffect(() => {
+        async function getAssets() {
+            //reset base values - needed so old data not shown
+            setAssetType({tags: [], thumnails: [], files: [], fields: []});
+            setThisAsset({flagsblocks: {BLOCK:[], FLAG:[], COUNT:{BLOCK: 0, FLAG: 0}}});
+            //query api
+            const fetchedAssets = await Api("assets/list.php", {"assetTypes_id": type,"all": true}, cancelToken.token);
+            try {
+                setAssetType(fetchedAssets['assets'][0]);//Store asset type
+
+                //find individual asset in list
+                for (let index = 0; index < fetchedAssets['assets'][0]['tags'].length; index++) {
```

Comment:
Could we map these instead?

Reply:
Would recommend we import underscore.js

Reply:
I did this to allow the loop to break when found, as map doesn't handle breaking. Have swapped back to map as is more efficient